packages/web/src/apollo/index.tsx

```diff
@@ -31,6 +31,35 @@ import { GraphQLHooksProvider } from '../components/GraphQLHooksProvider'
 
 export type ApolloClientCacheConfig = apolloClient.InMemoryCacheConfig
 
+export type RwLinkName =
+  | 'withToken'
+  | 'authMiddleware'
+  | 'updateDataApolloLink'
+  | 'httpLink'
+
+export type RwLink<
+  Name extends RwLinkName,
+  Link extends apolloClient.ApolloLink = apolloClient.ApolloLink
+> = Link & {
+  __rwLinkName: Name
+}
+
+function decorateLinkWithLinkName<
+  Name extends RwLinkName,
+  Link extends apolloClient.ApolloLink
+>(name: Name, link: Link): RwLink<Name, Link> {
+  const res = link as RwLink<Name, Link>
+  res.__rwLinkName = name
```

Comment:
I don't love mutating the Link here, but other options are not only more costly, but could also break reference to `this` since Links are class instances. Obviously there are workarounds but in this instance it makes sense to KISS